src/core/defs.jl

```diff
@@ -191,26 +203,38 @@ function set_dimension!(md::ModelDef, name::Symbol, keys::Union{Int, Vector, Tup
     return dim
 end
 
-# helper function for setindex; used to determine if the provided time values are a uniform range.
+# LFR-TBD:  We might be able to simplify this function, and we should think through
+# it's return carefully as it is at the root of several functions.  Also, do we
+# want to be able to call this on a model too (like years_array?) ... otherwise
+# we often call years_array and then isuniform.
+#
+# helper function for setindex; used to determine if the provided time values are 
+# a uniform range. The function returns -1 if the vector is not uniform, 
+#otherwise it returns the timestep length aka stepsize
 function isuniform(values::Vector)
```

Comment:
You can simplify this to
```
all_equal(values) = all(map(val -> val == values[1], values[2:end]))

# I'd keep this generic since it's useful, and for clarity, is* functions should return booleans
isuniform(values) = all_equal(diff(values))

# A function to check length(values), etc., and return -1 if variable or the step length if fixed.
function get_timestep(values) 
   ... 
end
```

Reply:
Done, with a few modifications to handle edge cases.